src/daemon/common_image_host.cpp

```diff
@@ -87,3 +91,37 @@ void mp::CommonVMImageHost::on_manifest_update_failure(const std::string& detail
     need_extra_update = true;
     mpl::log(mpl::Level::warning, category, fmt::format("Could not update manifest: {}", details));
 }
+
+void mp::CommonVMImageHost::check_remote_is_supported(const std::string& remote_name) const
+{
+    if (!MP_PLATFORM.is_remote_supported(remote_name))
+        throw mp::UnsupportedRemoteException(
+            fmt::format("Remote \'{}\' is not a supported remote for this platform. Please use "
+                        "`multipass find` for supported remotes and images.",
+                        remote_name));
+}
+
+void mp::CommonVMImageHost::check_alias_is_supported(const std::string& alias, const std::string& remote_name) const
+{
+    if (!MP_PLATFORM.is_alias_supported(alias, remote_name))
+        throw mp::UnsupportedAliasException(fmt::format(
+            "\'{}\' is not a supported alias. Please use `multipass find` for supported image aliases.", alias));
+}
+
+bool mp::CommonVMImageHost::check_all_aliases_are_supported(const QStringList& aliases,
+                                                            const std::string& remote_name) const
+{
+    for (const auto& alias : aliases)
+    {
+        try
+        {
+            check_alias_is_supported(alias.toStdString(), remote_name);
+        }
+        catch (const mp::UnsupportedAliasException&)
+        {
+            return false;
+        }
+    }
+
+    return true;
```

Comment:
You mean use this in lieu of the new `UnsupportedAliasException` type and get rid of `UnsupportedAliasException` all together, correct?

Reply:
Up to you. I thought that the symmetry between the checks throwing their own exception each was nice, and since you went to the trouble of creating the specific exception, you might as well keep it, in case we want to single out that specific problem in a future `catch`. But it's also ok if you want to remove it. As you prefer really.

Reply:
Ok, I think I'll leave it in place since as I like the symmetry as well.  On another note, I really need to remember all of these nice standard algorithms :smiley:

Reply:
Nice!  And done.